probe-rs/src/probe/jlink/mod.rs

```diff
@@ -255,6 +261,12 @@ impl ProbeFactory for JLinkFactory {
             // Assume the lowest value is a safe default
             _ => 504,
         };
+        this.config = this.read_device_config()?;
+        this.connection_handle = match selector.product_id {
+            // 0x1051: J-Link OB-K22-SiFive: reports "hardware fault or protocol violation"
+            0x1051 => None,
```

Comment:
I dislike adding exceptions, this should be an official j-link implementation, why does it behave vastly differently?

Reply:
Maybe we just don't hit this bit on legit HW? :)